libandroid-navigation/src/main/java/org/maplibre/navigation/android/navigation/v5/models/LegAnnotation.kt

```diff
@@ -0,0 +1,51 @@
+package org.maplibre.navigation.android.navigation.v5.models
+
+import kotlinx.serialization.Serializable
+
+/**
+ * An annotations object that contains additional details about each line segment along the route
+ * geometry. Each entry in an annotations field corresponds to a coordinate along the route
+ * geometry.
+ *
+ * @since 2.1.0
+ */
+@Serializable
+data class LegAnnotation(
+
+    /**
+     * The distance, in meters, between each pair of coordinates.
+     *
+     * @since 2.1.0
+     */
+    val distance: List<Double>?,
+
+    /**
+     * The speed, in meters per second, between each pair of coordinates.
+     *
+     * @since 2.1.0
+     */
+    val duration: List<Double>?,
+
+    /**
+     * The speed, in meters per second, between each pair of coordinates.
+     *
+     * @since 2.1.0
+     */
+    val speed: List<Double>?,
+
+    /**
+     * The posted speed limit, between each pair of coordinates.
+     * Maxspeed is only available for the `mapbox/driving` and `mapbox/driving-traffic`
+     * profiles, other profiles will return `unknown`s only.
+     *
+     * @since 3.0.0
+     */
+    val maxSpeed: List<MaxSpeed>?,
```

Comment:
Before maxSpeed was: 
`public abstract List<MaxSpeed> maxspeed();`

Reading this value from json doesn't work anymore.

Reply:
The Json looks like this:

<img width="172" alt="Screenshot 2024-12-03 at 09 50 35" src="https://github.com/user-attachments/assets/1cc4c2d3-c5ff-4866-9d91-fb48a01c61d8">

Reply:
Ahh, that was my bad. Thanks for finding this!